### PR TITLE
Add client debug features

### PR DESCRIPTION
**Pull request by lixmal**
## Describe your changes

- Adds `--anonymize` flag to `netbird status`
```
$ netbird status -dA

$ ./netbird status -dA --filter-by-status connected
[...]
OS: linux/amd64
Daemon version: development
CLI version: development
Management: Connected to https://api.stage.anon-Fp8km.domain:443
Signal: Connected to https://signal.stage.anon-Fp8km.domain:443
Relays:
  [stun:api.stage.anon-Fp8km.domain:3478] is Available
  [turn:api.stage.anon-Fp8km.domain:3478?transport=udp] is Available
Nameservers:
  [8.8.8.8:53, 8.8.4.4:53] for [.] is Available
FQDN: vik-x1-1.anon-9I9nz.domain
NetBird IP: 100.89.32.208/16
Interface type: Kernel
Quantum resistance: false
Routes: -
Peers count: 3/3 Connected

```

- Adds `netbird debug bundle -A` command to anonymize client logs and status output

```
$ netbird debug bundle -A
/tmp/netbird.debug.836717317.zip
```

- Formats `last handshake` and `last update` in a human friendly way (similar to the `wg` command)
```
$ netbird status

Peers detail:
 test-vpc-cloud-centos.netbird.stage:
  NetBird IP: 100.89.60.171
  Public key: rexh6U7uWqsqUGERfMqpkJGAgLzx9SRwXHEcM7xink4=
  Status: Connected
  -- detail --
  Connection type: P2P
  Direct: true
  ICE candidate (Local/Remote): srflx/prflx
  ICE candidate endpoints (Local/Remote): 198.51.100.0:61412/198.51.100.1:51820
  Last connection update: 6 minutes, 10 seconds ago
  Last WireGuard handshake: 13 seconds ago
  Transfer status (received/sent) 904 B/780 B
  Quantum resistance: false
  Routes: 10.0.1.0/24
  Latency: 37.557992ms
[...]
```

- Adds OS and ARCH to status output
```
$ netbird status -d

OS: linux/amd64
Daemon version: development
CLI version: development
[...]
```

- Adds log level selector
```
$ netbird debug log level trace

Log level set successfully to trace
```
 
- Adds debug bundle collector over time
```
$ netbird debug for 10s -A

Netbird down
Log level set to trace.
Netbird up
Remaining time: 00:00:01
Duration completed
Netbird down
Creating debug bundle...
/tmp/netbird.debug.3779269856.zip
```

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [x] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/cmd/debug.go [hunk 1]
```diff
@@ -12,34 +15,176 @@ import (
...
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := proto.NewDaemonServiceClient(conn)
+	level := parseLogLevel(args[0])
+	if level == proto.LogLevel_UNKNOWN {
+		cmd.Printf("Unknown log level: %s. Available levels are: panic, fatal, error, warn, info, debug, trace\n", args[0])
+		return nil
```
Comment (pappz) [inline client/cmd/debug.go]: I think this is an error situation. It should return with an error.

File: util/anonymize/anonymize.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
+package anonymize
```
Comment (pappz) [inline util/anonymize/anonymize.go]: Why did you place the anonymize into the root folder/util? In the client folder fits better.

**Review by pappz — CHANGES_REQUESTED** [review]

File: client/cmd/debug.go (hunk 1 above)
Comment (lixmal) [reply client/cmd/debug.go]: I didn't wanna clutter the output too much, but yeah, fixed

File: util/anonymize/anonymize.go (hunk 2 above)
Comment (lixmal) [reply util/anonymize/anonymize.go]: To be able to use it elsewhere, moved

File: client/server/debug.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		return nil, fmt.Errorf("create zip file: %w", err)
+	}
+	defer func() {
+		if err := bundlePath.Close(); err != nil {
+			log.Errorf("failed to close zip file: %v", err)
+		}
+
+		if err != nil {
+			if err2 := os.Remove(bundlePath.Name()); err2 != nil {
+				log.Errorf("Failed to remove zip file: %v", err)
```
Comment (pappz) [inline client/server/debug.go]: Should print err2 instead of err

Comment (lixmal) [reply client/server/debug.go]: fixed

**Review by pappz — APPROVED** [review]

**Pull request merged**